editoast/src/views/timetable.rs

```diff
@@ -593,47 +641,49 @@ fn build_conflict_core_request(
         );
     }
 
-    let mut occurrences = HashMap::new();
-
     // Build paced train requirements
-    for (train, (sim, _)) in paced_trains_ts.into_iter().zip(paced_train_simulations) {
-        let final_output = match sim {
-            SimulationResponse::Success { final_output, .. } => final_output,
-            _ => continue,
-        };
 
-        let entry = occurrences.entry(train.id).or_insert(0);
-        let occurrence_id = *entry;
-        let key = format!("{}#{}", train.id, occurrence_id);
-        map_string_to_id.insert(
-            key.clone(),
-            TrainId::PacedTrainOccurrence(PacedTrainOccurrenceId {
-                paced_train_id: train.id,
-                index: occurrence_id,
-            }),
-        );
+    let mut it = paced_train_simulations.into_iter();
+    for train in paced_trains {
+        let occurrences = &train.num_occurrences();
+        let items: Vec<_> = it.by_ref().take(*occurrences).collect();
```

Comment:
I don't think you need to `.collect()` here since you only need the count of it if I understand correctly.

```suggestion
        let items: Vec<_> = it.by_ref().take(*occurrences).count();
```

Reply:
thanks, done :)

Reply:
I forgot that an iterator can only be consumed once... I need `items` and I use it below